lib/gax.js

```diff
@@ -248,17 +249,198 @@ function BundleOptions(options) {
-  assert(typeof requestByteLimit !== 'number', 'should be a number');
-  assert(typeof delayThreshold !== 'number', 'should be a number');
-  assert(elementCountThreshold <= 0 && requestByteLimit <= 0 &&
-         delayThreshold <= 0, 'one threashold should be > 0');
+  assert(typeof elementCountThreshold === 'number', 'should be a number');
+  assert(typeof elementCountLimit === 'number', 'should be a number');
+  assert(typeof requestByteThreshold === 'number', 'should be a number');
+  assert(typeof requestByteLimit === 'number', 'should be a number');
+  assert(typeof delayThreshold === 'number', 'should be a number');
+  assert(elementCountThreshold > 0 || requestByteLimit > 0 ||
+         delayThreshold > 0, 'one threashold should be > 0');
   this.elementCountThreshold = elementCountThreshold;
   this.elementCountLimit = elementCountLimit;
   this.requestByteThreshold = requestByteThreshold;
   this.requestByteLimit = requestByteLimit;
   this.delayThreshold = delayThreshold;
 }
 exports.BundleOptions = BundleOptions;
+
+
+/**
+ * Helper for {@link constructSettings}
+ *
+ * @param {object} methodConfig - A dictionary representing a single
+ *   ``methods`` entry of the standard API client config file. (See
+ *   {@link constructSettings} for information on this yaml.)
+ * @param {BundleOptions|OPTION_INHERIT} methodRetryOverride -
+ *   If set to OPTION_INHERIT, the retry settings are derived from
+ *   method config. Otherwise, this parameter overrides
+ *   methodConfig.
+ * @returns An Executor that configures bundling, or null if this
+ *   method should not bundle.
+ */
+function constructBundling(methodConfig, methodBundlingOverride) {
+  if ('bundling' in methodConfig) {
+    if (methodBundlingOverride === OPTION_INHERIT) {
+      var params = methodConfig.bundling;
+      var options = new BundleOptions({
+        'elementCountThreshold': params.element_count_threshold,
+        'elementCountLimit': params.element_count_limit,
+        'requestByteThreshold': params.request_byte_threshold,
+        'requestByteLimit': params.request_byte_limit,
+        'delayThreshold': params.delay_threshold_millis});
+      /*
+       * TODO: comment-out when bundling is supported.
+       * return new Executor(options)
+       */
+    } else if (methodBundlingOverride) {
+        /* return new Executor(method_bundling_override) */
+    }
+  }
+  return null;
+}
+
+/**
+ * Helper for {@link constructSettings}
+ *
+ * @param {object} methodConfig - A dictionary representing a single
+ *   ``methods`` entry of the standard API client config file. (See
+ *   {@link constructSettings} for information on this yaml.)
+ * @param {RetryOptions|OPTION_INHERIT} methodRetryOverride -
+ *   If set to OPTION_INHERIT, the retry settings are derived from
+ *   method config. Otherwise, this parameter overrides
+ *   ``method_config``.
+ * @param {object?} retryCodes - A dictionary parsed from the
```

Comment:
I think the '?' comes before the type in JSDoc, but not positive: http://usejsdoc.org/tags-type.html